packages/react/src/utilities/positioning/positioning.ts

```diff
@@ -513,7 +513,14 @@ function _finalizeElementPosition(
   const hostRect: Rectangle = _getRectangleFromElement(hostElement);
   const elementEdge = coverTarget ? targetEdge : targetEdge * -1;
   let returnEdge = alignmentEdge ? alignmentEdge : _getFlankingEdges(targetEdge).positiveEdge;
-  if (!doNotFinalizeReturnEdge) {
+
+  // If the opposite edge is flush against the bounds,
+  // choose that as the anchor edge so the element rect can grow away from the bounds' edge
+  const oppositeReturnEdge = getOppositeEdge(returnEdge);
+  const oppositeEdgeOnBounds =
+    bounds && _getEdgeValue(elementRectangle, oppositeReturnEdge) === _getEdgeValue(bounds, oppositeReturnEdge);
```

Comment:
nitpick: This block of code can be avoided if `doNotFinalizeReturnEdge` is `false`.